### PR TITLE
Major refactor.

### PR DESCRIPTION
**Pull request by maximelb**
## Description of the change

Major refactor of USP and simplification.
- New connection header format.
- Removed framing.

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -334,10 +304,7 @@ func (c *Client) keepAliveSender() {
 		}
 
 		c.conn.SetWriteDeadline(time.Now().Add(5 * time.Second))
-		if err := c.conn.WriteJSON(frame{
-			ModuleID: moduleIDUSP,
-			Messages: []interface{}{},
-		}); err != nil {
+		if err := c.conn.WriteJSON(map[string]interface{}{}); err != nil {
```
Comment (mireaulf) [inline client.go:307]: The keepalive is completely empty?

File: messages.go [hunk 2]
```diff
@@ -4,22 +4,27 @@ const (
 	uspControlMessageACK       = "ack"
 	uspControlMessageBACKOFF   = "bo"
 	uspControlMessageRECONNECT = "re"
+	uspControlMessageERROR     = "error"
```
Comment (mireaulf) [inline messages.go:7]: I noticed we don't do anything for error messages?

File: client.go (hunk 1 above)
Comment (maximelb) [reply client.go:307]: Yup, just to test the connection and prevent positive confirmation we're still there.

File: messages.go (hunk 2 above)
Comment (maximelb) [reply messages.go:7]: Good catch, I had something for it but it got lost in a merge. Will add.

**Pull request merged**